cpp/include/cudf/detail/groupby/sort_helper.hpp

```diff
@@ -132,7 +132,10 @@ struct sort_groupby_helper {
   /**
    * @brief Get the number of groups in `keys`
    */
-  size_type num_groups() { return group_offsets().size() - 1; }
+  size_type num_groups(rmm::cuda_stream_view stream = rmm::cuda_stream_default)
```

Comment:
I think we should avoid defaulting streams as much as possible to force the caller to provide a stream.

Reply:
done.